### PR TITLE
libgourou: init at 0.8.1

### PR DESCRIPTION
**Pull request by sevenautumns**
###### Description of changes

[libgourou](https://indefero.soutade.fr/p/libgourou/) is a library for removing DRM from Adobe ebooks.  
This superseeds the stale #179678 pr, and requires less patches

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libgourou/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{ lib, stdenv, fetchurl, pugixml, updfparser, curl, openssl, libzip, unzip }:
+
+stdenv.mkDerivation rec {
+  name = "libgourou";
+  version = "0.8.1";
+
+  src = fetchurl {
```
Comment (dadada) [inline pkgs/development/libraries/libgourou/default.nix:7]: You can use `fetchzip` here.

Comment (sevenautumns) [reply pkgs/development/libraries/libgourou/default.nix:7]: I have tried that before and unfortunately that doesn't result in automatic unpacking either.

> trying https://indefero.soutade.fr/p/libgourou/source/download/v0.8.1/
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 100 80611    0 80611    0     0  29750      0 --:--:--  0:00:02 --:--:-- 29756
> unpacking source archive /build/v0.8.1
> do not know how to unpack source archive /build/v0.8.1

Comment (sevenautumns) [reply pkgs/development/libraries/libgourou/default.nix:7]: `fetchgit` also does not work, because apparently soutade.fr requires ssh for cloning

Comment (dadada) [reply pkgs/development/libraries/libgourou/default.nix:7]: Try `extension = "zip"`. Also, the download resulted in a different hash for me. I have `sha256-ZKnA+Ip96G1Iy6EDqu6BIz8O90MfqOUbla0z2RNSbNI=`

**Review by dadada — APPROVED** [review]
Sadly I don't own any ADEPT DRM protected files, so I can't confirm all binaries are working as intended. The PR looks fine otherwise.

**Comment by sevenautumns** [comment]
> Sadly I don't own any ADEPT DRM protected files, so I can't confirm all binaries are working as intended. The PR looks fine otherwise.

This can easily be tested with adobe's [free sample books](https://www.adobe.com/solutions/ebook/digital-editions/sample-ebook-library.html)  
Here simply download any ebook provided as an `acsm` file. 
(For example "The Adventures of Sherlock Holmes")

**Comment by sevenautumns** [comment]
@thiagokokada  
Since you reviewed the preceding #179678 pr, could I ask you for a review of this pr as well?

File: pkgs/development/libraries/libgourou/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    install -Dt $out/include include/libgourou*.h
+    install -Dt $out/lib libgourou.so
+    install -Dt $out/lib libgourou.so.${version}
+    install -Dt $out/lib libgourou.a
+    install -Dt $out/bin utils/acsmdownloader
+    install -Dt $out/bin utils/adept_{activate,loan_mgt,remove}
+    install -Dt $out/share/man/man1 utils/man/*.1
```
Comment (thiagokokada) [inline pkgs/development/libraries/libgourou/default.nix:36]:
```suggestion
    installManPage utils/man/*.1
```

Needs also `installShellFiles` in `nativeBuildInputs`.

File: pkgs/development/libraries/updfparser/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchzip {
+    url = "https://indefero.soutade.fr/p/updfparser/source/download/${rev}/";
+    sha256 = "sha256-Kt1QDj7E0GaT615kJW2MQKF9BeU5U7/95TQKODpxgNI=";
+    extension = "zip";
+  };
+
+  buildPhase = ''
+    make BUILD_STATIC=1 BUILD_SHARED=1
+  '';
```
Comment (thiagokokada) [inline pkgs/development/libraries/updfparser/default.nix:16]:
```suggestion
  makeFlags = [
    "BUILD_STATIC=1"
    "BUILD_SHARED=1"
  ];
```

File: pkgs/development/libraries/libgourou/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  postPatch = ''
+    patchShebangs scripts/setup.sh
+  '';
+
+  configurePhase = ''
+    mkdir lib
+    ln -s ${updfparser}/lib lib/updfparser
+  '';
```
Comment (thiagokokada) [inline pkgs/development/libraries/libgourou/default.nix:20]: Does something like this works instead?
```suggestion
  makeFlags = [
    "UPDFPARSERLIB=${updfparser}/lib"
  ];
```

**Review by thiagokokada — COMMENTED** [review]
Please introduce each package in separate commits:

```
updfparser: init at unstable-2023-01-10
libgourou: init at 0.8.1
```

File: pkgs/development/libraries/libgourou/default.nix (hunk 4 above)
Comment (thiagokokada) [reply pkgs/development/libraries/libgourou/default.nix:20]: No it doesn't.

Comment (thiagokokada) [inline pkgs/development/libraries/libgourou/default.nix:20]:
```suggestion
  postConfigure = ''
    mkdir lib
    ln -s ${updfparser}/lib lib/updfparser
  '';
```

**Comment by sevenautumns** [comment]
Thanks for the review @thiagokokada. I've applied all of them while some look a little different due to my formatter.

File: pkgs/development/libraries/updfparser/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, fetchzip }:
+
+stdenv.mkDerivation rec {
+  name = "updfparser";
+  version = "unstable-2023-01-10";
+  rev = "a421098";
```
Comment (thiagokokada) [inline pkgs/development/libraries/updfparser/default.nix:6]:
```suggestion
  rev = "a421098092ba600fb1686a7df8fc58cd67429f59";
```

Comment (sevenautumns) [reply pkgs/development/libraries/updfparser/default.nix]: done

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**